homeassistant/components/bring/todo.py

```diff
@@ -0,0 +1,157 @@
+        """Return the todo items."""
+        return [
+            TodoItem(
+                uid=item["name"],
+                summary=item["name"],
+                description=item["specification"] or "",
+                status=TodoItemStatus.NEEDS_ACTION,
+            )
+            for item in self.list["items"]
+        ]
+
+    async def async_create_todo_item(self, item: TodoItem) -> None:
+        """Add an item to the To-do list."""
+        try:
+
+            def _sync():
+                return self.coordinator.bring.saveItem(
+                    self.unique_id, item.summary, item.description or ""
+                )
+
+            await self.hass.async_add_executor_job(_sync)
+        except BringRequestException as e:
+            _LOGGER.warning("Unable to save todo item for bring")
+            raise HomeAssistantError from e
+
+        await self.async_update_ha_state(force_refresh=True)
+
+    async def async_update_todo_item(self, item: TodoItem) -> None:
+        """Update an item to the To-do list.
+
+        Bring has an internal 'recent' list which we want to use instead of a todo list status, therefore completed todo list items will directly be deleted
+
+        This results in following behaviour:
+
+        - Completed items will move to the "completed" section in home assistant todo list and get deleted in bring, which will remove them from the home assistant todo list completely after a short delay
+        - Bring items do not have unique identifiers and are using the name/summery/title. Therefore the name is not to be changed! Should a name be changed anyway, a new item will be created instead and no update for this item is performed and on the next cloud pull update, it will get cleared
+        """
+
+        if TYPE_CHECKING:
+            assert item.uid
+
+        if item.status == TodoItemStatus.COMPLETED:
+            await self.async_delete_todo_items([item.uid])
+
+        elif item.summary == item.uid:
+            try:
+
+                def _sync():
+                    return self.coordinator.bring.updateItem(
+                        self.unique_id, item.uid, item.description or ""
+                    )
+
+                await self.hass.async_add_executor_job(_sync)
+            except BringRequestException as e:
+                _LOGGER.warning("Unable to update todo item for bring")
+                raise HomeAssistantError from e
+        else:
+            await self.async_delete_todo_items([item.uid])
+            await self.async_create_todo_item(item)
+
+        await self.coordinator.async_refresh()
+
+    async def async_delete_todo_items(self, uids: list[str]) -> None:
+        """Delete an item from the To-do list."""
+        for uid in uids:
+            try:
+
+                def _sync():
+                    return self.coordinator.bring.removeItem(self.unique_id, uid)  # noqa: B023
+
+                await self.hass.async_add_executor_job(_sync)
```

Comment:
Like adding in a separate function was only there to avoid having 2 executor jobs quickly behind each other